src/main/scala/scala/compat/java8/collectionImpl/Accumulator.scala

```diff
@@ -0,0 +1,344 @@
+    }
+    totalSize += that.totalSize
+    that.clear
+  }
+  
+  override def clear(): Unit = {
+    super.clear()
+    current = Accumulator.emptyAnyRefArray
+    history = Accumulator.emptyAnyRefArrayArray
+    cumul  = Accumulator.emptyLongArray
+  }
+  
+  /** Retrieves the `ix`th element. */
+  final def apply(ix: Long): A = {
+    if (totalSize - ix <= index || hIndex == 0) current((ix - (totalSize - index)).toInt).asInstanceOf[A]
+    else {
+      val w = seekSlot(ix)
+      history((w >>> 32).toInt)((w & 0xFFFFFFFFL).toInt).asInstanceOf[A]
+    }
+  }
+  
+  /** Retrieves the `ix`th element, using an `Int` index. */
+  final def apply(i: Int): A = apply(i.toLong)
+  
+  /** Returns a `Stepper` over the contents of this `Accumulator`*/
+  final def stepper: AnyStepper[A] = new AccumulatorStepper[A](this)
+  
+  /** Returns an `Iterator` over the contents of this `Accumulator`. */
+  final def iterator = stepper.iterator
+
+  /** Returns a `java.util.Spliterator` over the contents of this `Accumulator`*/
+  final def spliterator: java.util.Spliterator[A] = stepper.spliterator
+  
+  /** Produces a sequential Java 8 Stream over the elements of this `Accumulator`*/
+  final def seqStream: java.util.stream.Stream[A] = java.util.stream.StreamSupport.stream(spliterator, false)
+  
+  /** Produces a parallel Java 8 Stream over the elements of this `Accumulator`*/
+  final def parStream: java.util.stream.Stream[A] = java.util.stream.StreamSupport.stream(spliterator, true)
+  
+  /** Copies the elements in this `Accumulator` into an `Array` */
+  final def toArray(implicit tag: reflect.ClassTag[A]) = {
+    if (totalSize > Int.MaxValue) throw new IllegalArgumentException("Too many elements accumulated for an array: "+totalSize.toString)
+    val a = new Array[A](totalSize.toInt)
+    var j = 0
+    var h = 0
+    var pv = 0L
+    while (h < hIndex) {
+      val x = history(h)
+      val n = cumulative(h) - pv
+      pv = cumulative(h)
+      var i = 0
+      while (i < n) {
+        a(j) = x(i).asInstanceOf[A]
+        i += 1
+        j += 1
+      }
+      h += 1
+    }
+    var i = 0
+    while (i < index) {
+      a(j) = current(i).asInstanceOf[A]
+      i += 1
+      j += 1
+    }
+    a
+  }
+  
+  /** Copies the elements in this `Accumulator` to a `List` */
+  final def toList: List[A] = {
+    var ans: List[A] = Nil
+    var i = index - 1
+    while (i >= 0) {
+      ans = current(i).asInstanceOf[A] :: ans
+      i -= 1
+    }
+    var h = hIndex - 1
+    while (h >= 0) {
+      val a = history(h)
+      i = (cumulative(h) - (if (h == 0) 0L else cumulative(h-1))).toInt - 1
+      while (i >= 0) {
+        ans = a(i).asInstanceOf[A] :: ans
+        i -= 1
+      }
+      h -= 1
+    }
+    ans
+  }
+  
+  /** Copies the elements in this `Accumulator` to a specified collection.
+    * Usage example: `acc.to[Vector]`
+    */
+  final def to[Coll[_]](implicit cbf: collection.generic.CanBuildFrom[Nothing, A, Coll[A]]): Coll[A] = {
```

Comment:
Could this be moved up to `AccumulatorLike`? Scala specialization should be able to produce the correct specialized versions for the subclasses automatically.